modules/affile/affile-dest.c

```diff
@@ -354,6 +358,29 @@ affile_dw_new(const gchar *filename, GlobalConfig *cfg)
   return self;
 }
 
+static void
+affile_dw_reopen_writer(gpointer key, gpointer value, gpointer user_data)
+{
+  AFFileDestWriter *writer = (AFFileDestWriter *) value;
+  affile_dw_reopen(writer);
+}
+
+static void
+affile_dd_reopen_all_writers(gpointer data, gpointer user_data)
+{
+  AFFileDestDriver *driver = (AFFileDestDriver *) data;
+  if (driver->single_writer)
+    affile_dw_reopen(driver->single_writer);
+  else
```

Comment:
I am sorry, but I think I might have done an incorrect assumption in this logic. That a destination file driver must either have single_writer or writer_hash (in case destination is a template) set.

I was just testing this patch again with a more complex configuration and it seems that sometimes both entries could be NULL as I got this message: `g_hash_table_foreach: assertion 'hash_table != NULL' failed`

In what configuration scenario would this happen and what is the proper logic? An `else if` or just another `if` (i.e. could both be unset or set)? @bazsi @furiel

Reply:
As I see the construction of single_writer or writer_hash is deferred until a file needs to be opened. And opening a file is deferred until there is some message to write.
Probably when syslog-ng starts and there was no message to handle, both these objects can be null.

Reply:
Ah yes, that makes sense. So an `else if` on the `writer_hash` should account for this.  
I'll create an issue of it and another pull request to solve it. Thanks!